dialogue-core/src/main/java/com/palantir/dialogue/core/RetryingChannel.java

```diff
@@ -182,29 +195,35 @@ private long getBackoffNanoseconds() {
 
         ListenableFuture<Response> handleHttpResponse(Response response) {
             if (Responses.isQosStatus(response)) {
-                if (++failures <= maxRetries) {
-                    response.close(); // nobody is going to read this response body
-                    Throwable throwableToLog = log.isInfoEnabled()
-                            ? new SafeRuntimeException(
-                                    "Received retryable response", SafeArg.of("status", response.code()))
-                            : null;
-                    return scheduleRetry(throwableToLog);
-                }
-                if (log.isDebugEnabled()) {
-                    log.debug(
-                            "Exhausted {} retries, returning a retryable response with status {}",
-                            SafeArg.of("retries", maxRetries),
-                            SafeArg.of("status", response.code()));
-                }
-                // not closing the final response body because ConjureBodySerde needs to read it to deserialize
-                return Futures.immediateFuture(response);
+                return incrementFailuresAndMaybeRetry(response, qosThrowable);
+            }
+
+            if (Responses.isServerError(response) && safeAndIdempotent(endpoint.httpMethod())) {
```

Comment:
plan is to stick to 500s initially

Reply:
Also possibly want a meter so we can watch a graph and see how often this fires.